src/VoltageCalibration.cc

```diff
@@ -1058,110 +1053,42 @@ double * mattak::applyVoltageCalibration (int N, const int16_t * in, double * ou
     return 0;
   }
 
-  bool is2ndHalfWindows;
-  int nwindows = N / mattak::k::radiant_window_size;
-  int isamp;
-  int i = 0;
-  int j;
+  int nSamples_wf = N;
+  int nSamplesPerGroup = mattak::k::num_radiant_samples;
+  int nWindowsPerGroup = mattak::k::radiant_windows_per_buffer;
+  int isamp_lab4, isamp_A, isamp_B;
 
-  if (start_window > 15)
+  if (isOldFirmware)
   {
-    is2ndHalfWindows = true;
-    j = start_window - mattak::k::radiant_windows_per_buffer;
+    nSamplesPerGroup /= 2;
+    nWindowsPerGroup /= 2;
+    isamp_A = (start_window / nWindowsPerGroup) * nSamplesPerGroup;
   }
   else
   {
-    is2ndHalfWindows = false;
-    j = start_window;
+    isamp_A = (start_window >= nWindowsPerGroup) * nSamplesPerGroup;
   }
 
-
-  for (int iwindow = 0; iwindow < nwindows; iwindow++)
+  for (int i = 0; i < nSamples_wf; i++)
   {
-    if (isOldFirmware)
-    {
-      // Old Firmware
-      // Wrap around if j % 16 = 0
-      if (j % mattak::k::radiant_windows_per_buffer == 0) j = 0;
-      if (inRange(0,4,j) || inRange(8,12,j)) isamp = (j+3) * mattak::k::radiant_window_size;
-      if (inRange(5,7,j) || inRange(13,15,j)) isamp = (j-5) * mattak::k::radiant_window_size;
-    }
-    else
-    {
-      // New Firmware
-      // Wrap around if j % 16 = 0
-      if (j % mattak::k::radiant_windows_per_buffer == 0) j = 0;
-      if (inRange(0,12,j)) isamp = (j+3) * mattak::k::radiant_window_size;
-      if (inRange(13,15,j)) isamp = (j-13) * mattak::k::radiant_window_size;
-    }
+    isamp_B = (i + start_window * mattak::k::radiant_window_size) % nSamplesPerGroup;
 
-    if (is2ndHalfWindows) isamp += mattak::k::num_radiant_samples;
+    isamp_lab4 = isamp_A + isamp_B;
 
-    j++;
+    const double *params = packed_fit_params + isamp_lab4 * (fit_order+1);
```

Comment:
@martin01021207 please re-write this line as suggested in slack for more readability. You could also add a comment briefly explaining it.

Reply:
I mean besides adding some comments what should I change here?